java-client/src/test/java11/org/opensearch/client/opensearch/integTest/AbstractGenericClientIT.java

```diff
@@ -0,0 +1,225 @@
+                                + "                }"
+                                + "            ]"
+                                + "        }"
+                                + "    },"
+                                + "    \"_source\": true\n"
+                                + "}"
+                        )
+                        .build()
+                )
+        ) {
+            assertThat(response.getStatus(), equalTo(200));
+            assertThat(response.getBody().isPresent(), equalTo(true));
+
+            final SearchResponse<?> r = Bodies.json(
+                response.getBody().get(),
+                SearchResponse.createSearchResponseDeserializer(JsonpDeserializer.of(ShopItem.class)),
+                javaClient()._transport().jsonpMapper()
+            );
+            assertEquals(r.hits().hits().size(), 2);
+
+            assertTrue(
+                Arrays.stream(r.hits().hits().get(0).fields().get("name").to(String[].class))
+                    .collect(Collectors.toList())
+                    .contains("hummer")
+            );
+
+            assertTrue(
+                Arrays.stream(r.hits().hits().get(1).fields().get("name").to(String[].class))
+                    .collect(Collectors.toList())
+                    .contains("jammer")
+            );
+        }
+    }
+
+    private void createTestDocuments(String index) throws IOException {
+        createTestDocument(index, "1", createItem("hummer", "huge", "yes", 2));
+        createTestDocument(index, "2", createItem("jammer", "huge", "yes", 1));
+        createTestDocument(index, "3", createItem("hammer", "large", "yes", 3));
+        createTestDocument(index, "4", createItem("drill", "large", "yes", 3));
+        createTestDocument(index, "5", createItem("jack", "medium", "yes", 2));
+        createTestDocument(index, "6", createItem("wrench", "medium", "no", 3));
+        createTestDocument(index, "7", createItem("screws", "small", "no", 1));
+        createTestDocument(index, "8", createItem("nuts", "small", "no", 2));
+    }
+
+    private void createTestDocument(String index, String id, ShopItem document) throws IOException {
+        try (
+            Response response = javaClient().generic()
+                .execute(
+                    Requests.builder(javaClient()._transport().jsonpMapper())
+                        .endpoint("/" + index + "/_doc/" + id)
+                        .method("PUT")
+                        .json(document)
+                        .build()
+                )
+        ) {
+            assertThat(response.getStatus(), equalTo(201));
+            assertThat(response.getBody().isPresent(), equalTo(true));
+        }
+    }
+
+    private void createIndex(String index) throws IOException {
+        try (
+            Response response = javaClient().generic()
+                .execute(
+                    Requests.builder(javaClient()._transport().jsonpMapper())
```

Comment:
Is there to lazy load this given that we will call `.json` later?

Reply:
Yes, sure, I could move it off to `.json()` method call